### PR TITLE
add foreman_global_parameter type

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

**Comment by jhoblitt** [comment]
I have removed the left over inclusion of `pry` from the spec.

File: lib/puppet/provider/foreman_global_parameter/rest_v3.rb [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    # debug("exists? resource: #{resource.to_hash}")
+    @property_hash[:ensure] == :present
+  end
+
+  def destroy
+    @property_hash[:ensure] = :absent
+  end
+
+  def flush
+    debug("flush @property_hash: #{@property_hash}")
```
Comment (alexjfisher) [inline lib/puppet/provider/foreman_global_parameter/rest_v3.rb:41]: Probably don't need this anymore??

File: lib/puppet/provider/foreman_global_parameter/rest_v3.rb [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  def id
+    @property_hash[:id]
+  end
+
+  def base_path
+    'api/v2/common_parameters'
+  end
+
+  def get_common_parameter!
+    debug('Getting')
```
Comment (alexjfisher) [inline lib/puppet/provider/foreman_global_parameter/rest_v3.rb:69]: Could you remove any of the debug statements that were only needed during development, or at least make them more descriptive than this?

Comment (jhoblitt) [reply lib/puppet/provider/foreman_global_parameter/rest_v3.rb]: I've removed all the debug statement.

**Comment by jhoblitt** [comment]
rebased on current master.

**Comment by jhoblitt** [comment]
rebased on current master.

**Comment by jhoblitt** [comment]
rebased on current master.

**Comment by jhoblitt** [comment]
rebased on current master.

**Review by alexjfisher — APPROVED** [review]

**Pull request merged**